### PR TITLE
Task/zuul content type cleaning

### PR DESCRIPTION
**Pull request by wtritch**
This adds a Zuul pre-filter that is a little more forgiving of slightly invalid Content-Type headers with an extra space.

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static com.netflix.zuul.context.RequestContext.getCurrentContext;
+
+/**
+ * This filter is responsible for cleaning extra whitespace from Content-Type
+ * header values.
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: can we add a link to the zuul spec here?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public int filterOrder() {
+        return 0;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        final RequestContext context = getCurrentContext();
+        return context.getRequest().getContentType() != null && context.getRequest().getContentType().contains(" ");
+    }
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: I only run the filter IFF a content-type header exists with a space in it.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    private static class ContentTypeRequestWrapper extends Servlet30RequestWrapper {
+
+        ContentTypeRequestWrapper(final HttpServletRequest request) {
+            super(request);
+        }
+
+        @Override
+        public String getContentType() {
+            return MediaType.valueOf(getRequest().getContentType()).toString();
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: Spring's MediaType implementation is much more forgiving of slightly invalid header values.

**Comment by jtreuting** [comment]
Is this only relevant on the load testing side, or would we expect this to impact normal usage?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public int filterOrder() {
+        return 0;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        final RequestContext context = getCurrentContext();
+        return context.getRequest().getContentType() != null && context.getRequest().getContentType().contains(" ");
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: should this be more inclusive?
like `/\s/g.test(contenttype)` ?

Comment (wtritch) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: hmmm for tabs and what-not?  Sounds reasonable.

**Comment by wtritch** [comment]
@jtreuting Since the test on whether or not to actually filter is very fast, and this is the second time I've run into this issue I think it's a reasonable "good thing" to add to the project permanently as opposed to a one-off "just for load-testing" hack.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    private static class ContentTypeRequestWrapper extends Servlet30RequestWrapper {
+
+        ContentTypeRequestWrapper(final HttpServletRequest request) {
+            super(request);
+        }
+
+        @Override
+        public String getContentType() {
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: Why do you need this helper class? Wouldn't Servlet30RequestWrapper get it done?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java (hunk 1 above)
Comment (wtritch) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: I'm not sure exactly what you mean?  What would you link to?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java (hunk 3 above)
Comment (malaffoon) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/zuul/RequestContentTypeCleaningFilter.java]: I really think we need a code comment in here somewhere. Something like, `This wrapper relies on MediaType to parse and format the content-type which might not strictly comply with the spec, which is what the zuul parser requires."

**Pull request merged**